### PR TITLE
Fix README typo + tweaks.

### PR DESCRIPTION
**Pull request by noffle**
s/interface-stream-muxer/interface-ipfs-core, and a couple of other small tweaks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -66,7 +67,7 @@ A valid (read: that follows this interface) IPFS core implementation, must expos
 
 `callback` must follow `function (err, node) {}` signature, where `err` is an error if the operation was not successful and `node` is a MerkleDAG node of the type [DAGNode][]
 
-If no `callback` is passed, a promise is returned.
+Returns a [promise][].
```
Comment (dignifiedquire) [inline README.md]: This change is not correct. The promise is only returned if no `callback` is passed.

Comment (noffle) [inline README.md]: Alas! I thought so too, but tried this with `promisify-es6` -- [it returns a promise regardless](https://gist.github.com/noffle/aac170fc16890c77809f0ba14d9fa235).

Comment (dignifiedquire) [inline README.md]: Interesting. Not sure if we want this though, I guess we have to fork it and make our own version in the long run.

**Comment by RichardLitt** [comment]
I'm liking the aggressive linking you're doing recently. +1.

File: README.md (hunk 1 above)
Comment (noffle) [inline README.md]: @dignifiedquire I think I agree about it being undesirable to always return a promise. A counterargument though might be that *lots* of people are already depending on promisify and us deviating from "the norm" may be detrimental to the larger community. Hm--I'm not sure. ping @diasdavid @nginnever too :)

**Comment by noffle** [comment]
@RichardLitt man, ever since I learned about markdown footnote links they've soo been my jam :heart:

File: README.md (hunk 1 above)
Comment (diasdavid) [inline README.md]: I would rather for it to be one or the other. Why can't we reach out to the dev of promisify-es6 and understand the reason behind his decision? Seems like the dev might have expected that if someone passes a callback, to not grab the promise as well, there would be nothing good coming from that.

Comment (diasdavid) [inline README.md]: Nevertheless, as for the spec. Let's rephrase to:

`A promise is also returned, you can skip passing the callback to the function.`

Comment (noffle) [inline README.md]: +1 for reaching out to the dev first -- this is how open source works best!

**Comment by noffle** [comment]
I reverted the phrasing back to how it was before for promises, but linkified them.

**Comment by diasdavid** [comment]
thank you @noffle :)

**Pull request merged**